### PR TITLE
DTSRD-1886: internal endpoint to retrieve organisation details

### PR DESCRIPTION
**Pull request by kiran-yenigala-hmcts**
### Checklist

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

**Review by lukasz-wolski — APPROVED** [review]

**Review by andy-m-b — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/professionalapi/controller/internal/OrganisationInternalController.java [hunk 1]
```diff
@@ -638,4 +639,39 @@ public ResponseEntity<Object> updateAnOrganisationsRegisteredPbas(
...
+    )
+    @ApiResponse(
+            responseCode = "403",
+            description = "Forbidden Error: Access denied",
+            content = @Content
+    )
+    @ApiResponse(
+            responseCode = "500",
+            description = "Internal Server Error",
+            content = @Content
```
Comment (rajkatla-hmcts) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/controller/internal/OrganisationInternalController.java:665]: @kiran-yenigala-hmcts , missing status code 400 and 404 in rest api doc, pls add.

File: src/main/java/uk/gov/hmcts/reform/professionalapi/controller/internal/OrganisationInternalController.java [hunk 2]
```diff
@@ -638,4 +639,39 @@ public ResponseEntity<Object> updateAnOrganisationsRegisteredPbas(
...
+    )
+
+    @GetMapping(
+            path = "/orgDetails/{userId}",
+            produces = APPLICATION_JSON_VALUE
+    )
+    @Secured("prd-admin")
+    public ResponseEntity<OrganisationEntityResponse> retrieveOrganisationByUserId(
+            @PathVariable("userId") @Valid @NotNull @NotBlank String userId) {
```
Comment (rajkatla-hmcts) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/controller/internal/OrganisationInternalController.java:674]: annotations @Valid @NotNull @NotBlank are not effective , pls remove them. as service method validating.

**Review by rajkatla-hmcts — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/professionalapi/controller/internal/OrganisationInternalController.java (hunk 2 above)
Comment (kiran-yenigala-hmcts) [reply src/main/java/uk/gov/hmcts/reform/professionalapi/controller/internal/OrganisationInternalController.java]: done

File: src/main/java/uk/gov/hmcts/reform/professionalapi/controller/internal/OrganisationInternalController.java (hunk 1 above)
Comment (kiran-yenigala-hmcts) [reply src/main/java/uk/gov/hmcts/reform/professionalapi/controller/internal/OrganisationInternalController.java]: done

**Review by rajkatla-hmcts — APPROVED** [review]

**Review by lukasz-wolski — APPROVED** [review]

**Pull request merged**